File: practica_6/p6ej3.py
# // Escribir una función recursiva para contar la cantidad de vocales que posee una cadena de 
# // caracteres S.
# //  Ejemplo si S = 'Práctica de Algoritmos’ la función retorna 8

s = "hola como andasaa"
cant = 0
r = 0

def contarvoc(x, i = 0):
    global r
    if x[i] == "a":
        r+=1
    if i != cant -1:
        contarvoc(x, i +1)
    return

cant = len(s)
contarvoc(s)
print(r)

File: practica_6/p6ej3.c
// Escribir una función recursiva para contar la cantidad de vocales que posee una cadena de 
// caracteres S.
//  Ejemplo si S = 'Práctica de Algoritmos’ la función retorna 8

#include <stdio.h>
#include <string.h>
int l = 0;
int res = 0;

int contarvocal(char x[], int i){
    char lo = x[i];
    if(lo == 'a'){
        res += 1;
    }
    if(i <= l-1){
        contarvocal(x, i+1);
    }
    return res;
}

int main()
{
    char s[] = "Practica de algoritmos";
    int i = 0;
    l = strlen(s);
    printf("%d", contarvocal(s, i));
    
    return 0;
}